AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/ModelStorageBehavior.swift

```diff
@@ -44,3 +44,13 @@ protocol ModelStorageBehavior {
                          completion: DataStoreCallback<[M]>)
 
 }
+
+protocol ModelStorageErrorBehavior {
+    func shouldIgnoreError(error: DataStoreError) -> Bool
```

Comment:
do you think conforming classes may also need the operation type as additional parameter here? 
I wonder that we may need to treat an error differently based on the context where it originated

Reply:
what's the operation type you are referring to here?

Reply:
I was referring to the "operation types" defined by `ModelStorageBehavior` like `query`, `save`, `delete` (but I don't think we have an actual Swift type for those)

Reply:
ah got it, im not sure if the decision to ignore an error differs based on operation type. we can keep this in mind if the use case extends beyond constraint violation errors